### PR TITLE
refactor: Fix clang-tidy readability-const-return-type violations

### PR DESCRIPTION
**Pull request by maflcko**
This comes up during review, so instead of wasting review cycles on this, just enforce it via CI

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by MarcoFalke** [comment]
For docs: https://clang.llvm.org/extra/clang-tidy/checks/readability/const-return-type.html

**Comment by hebasto** [comment]
MSVC linker [error](https://api.cirrus-ci.com/v1/task/5351057543397376/logs/build.log):
```
libbitcoin_node.lib(validationinterface.obj) : error LNK2019: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const __cdecl RemovalReasonToString(enum MemPoolRemovalReason const &)" (?RemovalReasonToString@@YA?BV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBW4MemPoolRemovalReason@@@Z) referenced in function "public: __cdecl `public: void __cdecl CMainSignals::TransactionRemovedFromMempool(class std::shared_ptr<class CTransaction const > const &,enum MemPoolRemovalReason,unsigned __int64)'::`4'::<lambda_2>::operator()(void)const " (??R<lambda_2>@?3??TransactionRemovedFromMempool@CMainSignals@@QEAAXAEBV?$shared_ptr@$$CBVCTransaction@@@std@@W4MemPoolRemovalReason@@_K@Z@QEBA@XZ) [C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build_msvc\bitcoind\bitcoind.vcxproj]
```

**Comment by MarcoFalke** [comment]
Thanks, force pushed to remove `const` from `validationinterface.cpp` as well.

**Review by hebasto — APPROVED** [review]
ACK fa7767ae278baa7074735c9eec070d992d2884ff

**Comment by MarcoFalke** [comment]
rebased

**Review by stickies-v — APPROVED** [review]
utACK fa451d4b6

I verified that all the removed `const` qualifiers were top-level, and as such can be safely removed. Checking this with `clang-tidy` makes the review process more efficient.

**Review by hebasto — APPROVED** [review]
ACK fa451d4b60ee0538b3ea6b946740a64734b35b6d.

Maybe add a commit to address https://github.com/bitcoin/bitcoin/pull/26705#discussion_r1092977011 as well?

File: src/test/fuzz/util.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ size_t CallOneOf(FuzzedDataProvider& fuzzed_data_provider, Callables... callable
 template <typename Collection>
 auto& PickValue(FuzzedDataProvider& fuzzed_data_provider, Collection& col)
 {
-    const auto sz = col.size();
+    auto sz{col.size()};
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/test/fuzz/util.h:50]: Correct, `decltype(sz)` is `const ...`, which will be the return value of `ConsumeIntegralInRange`, and thus cause the `readability-const-return-type` fail

**Pull request merged**